frontend/src/components/widgets/ColorPicker/ColorPicker.test.tsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2018-2020 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { mount, ReactWrapper } from "enzyme"
```

Comment:
We should use shallow instead of mount , we don't need to test baseweb components.

Reply:
I'd love to use shallow as well but my poor understanding is that shallow won't render children. Since the colopicker is a children of a popover component, I have to use mount to render it in the test. Any suggestions to get around this problem?

Reply:
Amie, yes in order to solve that you could [this method](https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/renderProp.html) but let me send you a proper example around your code

Reply:
Could be something like this:
```typescript
it("supports hex shorthand", () => {
    const props = getProps()
    const wrapper = shallow(<ColorPicker {...props} />)

    wrapper.find(UIPopover).simulate("click")

    const ChromePickerWrapper = wrapper.find(UIPopover).renderProp("content")()

    // @ts-ignore
    ChromePickerWrapper.prop("onChangeComplete")({
      hex: "#333",
    })

    expect(wrapper.find(UIPopover).renderProp("content")().prop("color")).toEqual("#333")
  })
```

Reply:
ok, I tried using renderProp but it doesn't work when I want to test a function prop on the children component. For example in this test:

```
  it("supports hex shorthand", () => {
    // with renderProp
    colorPickerWrapper.renderProp("onChangeComplete")({ hex: "#333" })

    // this doesnt work either
    // colorPickerWrapper.prop("onChangeComplete")({
    //   hex: "#333",
    // })

    // in both cases UI received {hex: "#333"} but the shallow copy in the test are not being updated
    colorPickerWrapper.update()

    expect(colorPickerWrapper.prop("color")).toEqual("#333")
  })
```

Reply:
Did you try the same as I sent?

Reply:
Did you try my exact code?, it's passing for me:
<img width="1276" alt="Screen Shot 2020-04-24 at 1 22 52 PM" src="https://user-images.githubusercontent.com/631997/80234610-d518a080-862e-11ea-97bc-f7d3c76a1477.png">

Reply:
That works! Thank you so much!

Why is that if I replace **wrapper.find(UIPopover).renderProp("content")()** on line 112 with **ChromePickerWrapper**  the test would fail? Aren't they the same thing?

Reply:
Because its not being mounted, it's just a prop that you have to re-check the wrapper component gets updated but in order to check that content you have to re-check the renderProp